adam/utils/networkx_utils.py

```diff
@@ -27,3 +30,25 @@ def digraph_with_nodes_sorted_by(
         new_graph.add_edge(source, dest, **data)
 
     return new_graph
+
+
+def subgraph(graph: DiGraph, nodes: Iterable[Any]) -> DiGraph:
+    """
+    Get a Subgraph view of a a Digraph with node iteration order in a deterministic fashion
```

Comment:
Expand this to explain why it is necessary, and cite your source for the code you modified to make this.